File: DeepSpaceRuby/lib/EnemyStarShip.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Deepspace
  require_relative 'Loot'
  require_relative 'Damage'
  require_relative 'EnemyToUI'
  
  class EnemyStarShip
    attr_reader:ammoPower
    attr_reader:name
    attr_reader:shieldPower
    attr_reader:loot
    attr_reader:damage
    
    def initialize(n,a,s,l,d)
       @ammoPower=a
       @name=n
       @shieldPower=s
       @loot=l
       @damage=d
    end
    
    def self.newCopy(e)
      new(e.name, e.ammoPower, e.shieldPower, e.loot, e.damage)
    end
    
    def getUIversion
      EnemyToUI.new(self)
    end
    
    def protection
      @shieldPower
    end
    
    def fire
      @ammoPower
    end
    
    def receiveShot(shot)
      if shot >= @shieldPower
        return ShotResult::DONOTRESIST
      else
        return ShotResult::RESIST
      end
    end
    
    def to_s
      "Name:#{@name} \n AmmoPower:#{@ammoPower} \n ShieldPower:#{@shieldPower} \n Loot:#{@loot} \n Damage:#{@damage}"
    end
   
  end
end

File: DeepSpaceJava/src/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author samuel y Laura
 */
public class EnemyStarShip implements SpaceFighter {
    
    private float ammoPower;
    String name;
    float shieldPower;
    Loot loot;
    Damage damage;
    
    
   EnemyStarShip(String n, float a, float s, Loot l, Damage d){
       ammoPower=a;
       name=n;
       shieldPower=s;
       loot = l;
       damage=d;
   }
   
   EnemyStarShip(EnemyStarShip e){
       
       ammoPower=e.ammoPower;
       name=e.name;
       shieldPower=e.shieldPower;
       loot =e.loot;
       damage=e.damage;
   }
   
   public float fire(){
       return ammoPower;
   }
   
    public float getAmmoPower() {
        return ammoPower;
    }
    
     public String getName() {
        return name;
    }

    public float getShieldPower() {
        return shieldPower;
    }

    public Loot getLoot() {
        return loot;
    }

    public Damage getDamage() {
        return damage;
    }
   
    EnemyToUI getUIversion(){
       return new EnemyToUI(this);
    }
   
    public float protection(){
       return shieldPower;
    }
 
    public ShotResult receiveShot(float shot){
        if(shot>shieldPower){
            return ShotResult.DONOTRESIST;
        } else
            return ShotResult.RESIST;
    }

    @Override
    public String toString() {
        return "EnemyStarShip{" + "name=" + name + ", shieldPower=" + shieldPower + ", loot=" + loot + ", damage=" + damage + '}';
    }


    
    
}
